elephant/conversion.py

```diff
@@ -152,3 +155,704 @@ def binarize(spiketrain, sampling_rate=None, t_start=None, t_stop=None,
+    and minimal stop point.
+
+    Parameters
+    ----------
+    spiketrains: neo.SpikeTrain object, list or array of neo.core.SpikeTrain
+                 objects
+        List of neo.core SpikeTrain objects to extract `t_start` and
+        `t_stop` from.
+
+    Returns
+    -------
+    start : quantities.Quantity
+        Start point extracted from input :attr:`spiketrains`
+    stop : quantities.Quantity
+        Stop point extracted from input :attr:`spiketrains`
+    """
+    if isinstance(spiketrains, neo.SpikeTrain):
+        return spiketrains.t_start, spiketrains.t_stop
+    else:
+        start = max([elem.t_start for elem in spiketrains])
+        stop = min([elem.t_stop for elem in spiketrains])
+    return start, stop
+
+
+class Binned:
+    """
+    Class which calculates a binned spike train and provides methods to
+    transform the binned spike train to clipped or unclipped matrix.
+
+    A binned spike train represents the occurrence of spikes in a certain time
+    frame.
+    I.e., a time series like [0.5, 0.7, 1.2, 3.1, 4.3, 5.5, 6.7] is
+    represented as [0, 0, 1, 3, 4, 5, 6]. The outcome is dependent on given
+    parameter such as size of bins, number of bins, start and stop points.
+
+    A clipped matrix represents the binned spike train in a binary manner.
+    It's rows represent the number of spike trains
+    and the columns represent the binned index position of a spike in a
+    spike train.
+    The calculated matrix columns contain only ones, which indicate
+    a spike.
+
+    An unclipped matrix is calculated the same way, but its columns
+    contain the number of spikes that occurred in the spike train(s). It counts
+    the occurrence of the timing of a spike in its respective spike train.
+
+    Parameters
+    ----------
+    spiketrains : List of `neo.SpikeTrain` or a `neo.SpikeTrain` object
+        Object or list of `neo.core.SpikeTrain` objects to be binned.
+    binsize : quantities.Quantity
+        Width of each time bin.
+        Default is `None`
+    num_bins : int
+        Number of bins of the binned spike train.
+        Default is `None`
+    t_start : quantities.Quantity
+        Time of the first bin (left extreme; included).
+        Default is `None`
+    t_stop : quantities.Quantity
+        Stopping time of the last bin (right extreme; excluded).
+        Default is `None`
+    store_mat : bool
+        If set to **True** unclipped matrix will be stored in memory.
+        If set to **False** matrix will always be calculated on demand.
+        This boolean indicates that the unclipped matrix will be stored in
+        memory.
+        Default is False.
+
+    See also
+    --------
+    __convert_to_binned
+    spike_indices
+    mat_clipped
+    mat_unclipped
+
+    Notes
+    -----
+    There are four cases the given parameters must fulfill.
+    Each parameter must be a combination of following order or it will raise
+    a value error:
+    * t_start, num_bins, binsize
```

Comment:
Yes, it isn't documented, but more importantly there is also no explicit handling of this in the code as far as I can tell.